TritonSE/DFM-Sideline-Sidekick-App
# Dynamic medical emergency page

## Tracking Info

Resolves #9 

## Changes

Changed conditions section from static rendering of documents to dynamic rendering which relies on pulling IDs from the backend database to retrieve the documents and displaying them on your device screen.

Currently, a dummy home screen is used to trigger any changes

Update from Meeting 2-15: Recursive rendering has been implemented meaning that headers, subheaders, subsubheaders, etc. and associated plaintext strings or bulleted string lists (as specified by user headers) are rendered.

## Testing and Confirmation of Change

In order to test this code, be sure to follow pre-setup:
1. Backend has the `.env` file that connects to the database and `npm install` followed by `npm start` have been utilized
2. Use the following `.env` file set to reference documents from RohanTest database:
```
PORT=3001
# MONGODB_URI="mongodb://127.0.0.1:27017/todoList"
# FRONTEND_ORIGIN="http://localhost:3000"
DATABASE_URL=mongodb+srv://[DM Anthony or Eshaan or Rohan for this part if you don't have DB URL].mongodb.net/RohanTest?retryWrites=true&w=majority
```
3. You have inserted your ip address into the designated part of `requests.ts` (search "note to reviewers"). Make sure to use `ipconfig` for Windows or `ifconfig` for Mac/Linux.
![image](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/72675fcf-1904-4995-9e77-5c266a395817)
So in this case, I would insert `100.112.104.53` in the designated place after the `http://` part and before the `:3001` part
4. The device that you are executing code from and the device you are testing from are on the same Wifi network
5. You have run `npm install` and `npx expo start` in the frontend folder
6. Without following step 2, documents will not be retrievable which will likely cause an error
7. Without following steps 3 and 4, you will likely hit a network error

Steps for Testing:
1. Scan QR Code on Expo
2. Code is currently set to first test case as discussed in HomeScreen.tsx. Press "Navigate to Conditions Section" and play around with it.
3. Switch test cases by changing ID variable according to comments in dummy HomeScreen.tsx and saving your code
4. Make sure to navigate back and press  "Navigate to Conditions Section" to trigger the update.
5. If you have mongodb compass access, there is a good interface for adding more documents to RohanTest db, so you can make more test cases and add them in if you want - be sure to follow the schema format. If the test case demonstrates an important area of improvement that isn't discussed below, be sure to commit and push the test.

Testing  Screenshots:
Currently, the feature seems to be working relatively well on Android phone and iPad as shown in the screenshots below. 

![IMG_8501](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/c3e398c2-0cfa-4463-9433-a199a5393663)
![IMG_8502](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/d08e63b0-5134-4af0-aaee-ad4a71317fde)
![Screenshot_20240215_201650_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/4a71e3f0-cd92-4b2c-ab3e-808e407a9778)
![Screenshot_20240215_201659_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/81b58cc3-c803-4c13-8c85-2dec11b2d6e2)
![Screenshot_20240215_201712_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/39351c2a-b981-45a7-be62-a915bfc0d9a7)
![Screenshot_20240215_201816_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/34d9f4f7-a380-41e4-b871-945e2a424c0c)
![Screenshot_20240215_201823_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/94fae29e-e2e2-4e02-825d-794fa0f93018)
![Screenshot_20240215_201840_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/5d99eb0d-ba2e-41a6-819d-1c3038c07788)
![Screenshot_20240215_201843_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/b85cb376-56be-4203-a3b5-6d31aef02717)
![Screenshot_20240215_201859_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/af1c8e74-371a-4a0a-a4f3-88995585cf66)
![Screenshot_20240215_201902_Expo Go](https://github.com/TritonSE/DFM-Sideline-Sidekick-App/assets/45048779/f4e0c641-e2c2-4faf-b24f-d0c7ee5a2c53)



TODO: Points of Improvement
Dummy home page needs to be replaced by global search page
Fix styling space by adding padding below
There is some extra spacing below the title and above the first header that needs to be resolved
Potential resultant linter or package issues that may arise

Feel free to add commits and push as necessary

DM Anthony, Eshaan, or Rohan with questions
